src/utils/visualize.py

```diff
@@ -0,0 +1,109 @@
+import gym
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from librosa.feature import melspectrogram
+from torch.utils.tensorboard import SummaryWriter
+
+from ..env.array_voc_state import VocStateObsNames as ObsNames
+from ..models.abc.agent import Agent
+from ..models.abc.world import World
+
+
+def make_spectrogram_figure(
+    target: np.ndarray,
+    generated: np.ndarray,
+    predicted_generated: np.ndarray,
+) -> None:
+    fig, axes = plt.subplots(3, 1)
+    fig.tight_layout()
+    labels = {"xlabel": "timestamp", "ylabel": "Hz"}
+
+    data = {
+        "Target": target,
+        "Generated": generated,
+        "Predicted Generated": predicted_generated,
+    }
+    # show target mel spectrogram
+    for i, (title, spect) in enumerate(data.items()):
+        mappable = axes[i].imshow(spect, aspect="auto")
+        axes[i].set(**labels, title=title)
+        fig.colorbar(mappable, ax=axes[i])
+
+    return fig
+
+
+def visualize_model_approximation(
```

Comment:
デコレーターにするとシンプルでインデントを浅くできます。
```suggestion
@torch.no_grad()
def visualize_model_approximation(
```